### DISCUSSION TITLE
error about plotting

### DISCUSSION THREAD

**Discussion by minnieliao** [phoebe-project/phoebe2#446] [Questions]
Hello everyone, this is the first time I using Phoebe. But I have a problem about plotting light curve. Here is my code and error message.

> import phoebe
> b = phoebe.default_binary(contact_binary = True)
> #print(b.hierarchy)
> #b.set_value('teff', component = 'primary', value = 8000)
> b.set_value('teff', component = 'secondary', value = 5000)
> b.add_dataset('lc', times=phoebe.linspace(0,2,101))
> b.run_compute()
> b.plot(x = 'phases', show = True)

 **/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/phoebe/dependencies/autofig/figure.py:627: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
 plt.show()  # <-- blocking**

I use Jupyter to do my work and my notebook is 2013 MacBook pro. I had try some methods to fix this problem, but it still don't work. This problem had perplexed me for few days. Thank you.

**Comment by kecnry** [comment]
Hello!  It sounds like your installation of matplotlib is defaulting to a backend which can only write to a file but not show inline.  I'm just not sure if that's the fault of your matplotlib configuration or PHOEBE.  For now, you can get around this by passing `save='filename.png` to the plot call instead of `show=True` and viewing the resulting file, but let's also see if we can get it working for inline plots as it should be.

Can you please try the following in a _fresh_ jupyter notebook and let me know the output?

```
import matplotlib
print(matplotlib.get_backend())
```

and then whether the following does plot and show in the notebook:

```
import matplotlib.pyplot as plt
plt.plot([1,2,3], [1,2,3])
plt.show()
```

Thanks!
Kyle

**Reply by minnieliao** [reply]
Thank you very much! It can work if I save the picture directly.

And I try the code you gave me about getting backend. The output shows `agg`. But there is something weird, for the last few months I use Matplotlib to plot data from ztf. The code works all right, it can show inline. So, I don't know which one (matplotlib configuration or PHOEBE) is fault.

**Reply by kecnry** [reply]
ok, then you can also manually set the backend by doing `import matplotlib; matplotlib.use('TkAgg')`, for example, at the _top_ of your script (try both before and after importing phoebe).  You can see a list of the different [matplotlib backends here](https://matplotlib.org/2.0.2/faq/usage_faq.html#what-is-a-backend) - you'll need to choose one where you have the dependencies installed ('Qt4Agg' might be your best bet if 'TkAgg' doesn't work).

**Reply by minnieliao** [reply]
I use 'TkAgg' before importing phoebe. And it can work!
By the way, I also try to use 'Qt4Agg' and 'Qt5Agg', one of the method I try before, but both of them didn't work.
Thank you!
